### ISSUE TITLE
dune 1.0 fails to build merlin master due to variable form

### ISSUE THREAD

**Issue by avsm** [ocaml/dune#974]
This is trying to build https://github.com/avsm/platform with dune 1.0, which vendors merlin master:

```
$ make
cd vendor/lwt && ocaml src/util/configure.ml -use-libev false
cd vendor/markup && ocaml src/configure.ml
dune build --profile=release @install
File "vendor/merlin/src/frontend/jbuild", line 32, characters 21-49:
Error: Unknown form: ${ocaml-config:ccomp_type}
```

**Comment by ejgallego** [comment]
Certainly Dune `master` may be bugged, need to research more but I just got this from our CI:
```
#=== ERROR while installing sexplib.v0.10.0 ===================================#
# opam-version         1.2.2
# os                   linux
# command              jbuilder build -p sexplib -j 2
# path                 /root/.opamcache/4.06.1/build/sexplib.v0.10.0
# compiler             4.06.1
# exit-code            1
# env-file             /root/.opamcache/4.06.1/build/sexplib.v0.10.0/sexplib-4341-e77081.env
# stdout-file          /root/.opamcache/4.06.1/build/sexplib.v0.10.0/sexplib-4341-e77081.out
# stderr-file          /root/.opamcache/4.06.1/build/sexplib.v0.10.0/sexplib-4341-e77081.err
### stderr ###
#           sh src/pre_sexp.pp.ml (exit 126)
# (cd _build/default && /bin/sh -c '${PA_CPP} src/pre_sexp.ml') > _build/default/src/pre_sexp.pp.ml
# /bin/sh: 1: src/pre_sexp.ml: Permission denied
```

**Comment by rgrinberg** [comment]
This is fixed in #971. Keen eye though :)
